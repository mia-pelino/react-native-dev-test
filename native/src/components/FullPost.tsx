import React from 'react';
import { RouteProp } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from 'react-native';
import { RootStackParamList } from '../types/RootStackParamList';
import Markdown from 'react-native-markdown-display';

type FullPostRouteProp = RouteProp<RootStackParamList, 'FullPost'>;

type FullPostNavigationProp = StackNavigationProp<
  RootStackParamList,
  'FullPost'
>;

type Props = {
  route: FullPostRouteProp;
  navigation: FullPostNavigationProp;
};

const FullPost: React.FC<Props> = ({ route, navigation }) => {
  return (
    <ScrollView style={styles.fullPost}>
      <TouchableOpacity
        onPress={() => {
          navigation?.navigate('Home');
        }}
      >
        <Text style={styles.navigateBack}>{'<'} Return to List of Posts</Text>
      </TouchableOpacity>
      <View>
        <Text style={styles.title}>{route.params.title}</Text>
        <Markdown style={markdownStyle}>{route.params.body}</Markdown>
      </View>
    </ScrollView>
  );
};

export default FullPost;

const styles = StyleSheet.create({
  fullPost: {
    backgroundColor: '#000',
    paddingHorizontal: 20,
  },
  title: {
    color: '#A5FFD6',
    paddingTop: 20,
    paddingBottom: 15,
    fontWeight: 'bold',
    fontSize: 24,
    alignSelf: 'center',
  },
  body: {
    flex: 1,
    flexWrap: 'wrap',
    color: '#FFF',
    fontSize: 16,
    opacity: 1,
    paddingHorizontal: 10,
  },
  navigateBack: {
    paddingTop: 10,
    color: '#FFA69E',
  },
});

const markdownStyle = {
  body: {
    ...styles.body,
    fontSize: 14,
  },
  heading1: {
    ...styles.body,
    fontSize: 22,
  },
  heading2: {
    ...styles.body,
    fontSize: 20,
  },
};
