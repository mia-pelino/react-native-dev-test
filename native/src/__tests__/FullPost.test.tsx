import React from 'react';
import renderer, { create, act } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import FullPost from '../components/FullPost';

describe('<FullPost /> snapshots', () => {
  it('displays title and full post sans markdown characters', async () => {
    const tree = renderer
      .create(
        <FullPost
          route={{
            params: {
              title: 'curly fries',
              body: '#seasoned ##with ranch dipping sauce',
            },
          }}
        />
      )
      .toJSON();

    await act(async () => {
      expect(tree).toMatchSnapshot();
    });
  });

  it('displays blank full post when title and body are empty', async () => {
    const tree = renderer
      .create(
        <FullPost
          route={{
            params: {
              title: '',
              body: '',
            },
          }}
        />
      )
      .toJSON();

    await act(async () => {
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('interactivity', () => {
  it('makes call to navigate to post list when message is pressed', async () => {
    const mockNavigation = jest.fn();
    const wrapper = render(
      <FullPost
        route={{
          params: {
            title: 'hash browns',
            body: 'crispy with hot sauce',
          },
        }}
        navigation={{
          navigate: mockNavigation,
        }}
      />
    );

    fireEvent.press(wrapper.getByText('< Return to List of Posts'));

    expect(mockNavigation).toHaveBeenCalledTimes(1);
  });
});
