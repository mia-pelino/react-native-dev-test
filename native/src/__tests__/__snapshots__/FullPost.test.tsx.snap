// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullPost /> snapshots displays blank full post when title and body are empty 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#000",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFA69E",
            "paddingTop": 10,
          }
        }
      >
        &lt;
         Return to List of Posts
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#A5FFD6",
            "fontSize": 24,
            "fontWeight": "bold",
            "paddingBottom": 15,
            "paddingTop": 20,
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
            "opacity": 1,
            "paddingHorizontal": 10,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`<FullPost /> snapshots displays title and full post sans markdown characters 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#000",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFA69E",
            "paddingTop": 10,
          }
        }
      >
        &lt;
         Return to List of Posts
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#A5FFD6",
            "fontSize": 24,
            "fontWeight": "bold",
            "paddingBottom": 15,
            "paddingTop": 20,
          }
        }
      >
        curly fries
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
            "opacity": 1,
            "paddingHorizontal": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginTop": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={Object {}}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFF",
                    "fontSize": 14,
                  },
                  Object {},
                ]
              }
            >
              #seasoned ##with ranch dipping sauce
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
